pkgs/build-support/fetchgit/default.nix

```
@@ -66,7 +66,7 @@ lib.makeOverridable (
             # when rootDir is specified, avoid invalidating the result when rev changes
             append = if rootDir != "" then "-${lib.strings.sanitizeDerivationName rootDir}" else "";
           },
-          leaveDotGit ? deepClone || fetchTags,
+          leaveDotGit ? null,
```

Should we add documentation here to clarify that setting this to null defaults:


```suggestion
          # When null, will default to: `deepClone || fetchTags`
          leaveDotGit ? null,
```